### PR TITLE
Add arm64 support to cpp mariadb container

### PR DESCRIPTION
**Pull request by dnastrain**
FYI, full description is over on the ["Public PR"](https://github.com/microsoft/vscode-dev-containers/pull/1328), so didn't add same text here...

### CODE REVIEW

File: containers/cpp-mariadb/.devcontainer/install-mariadb.sh [hunk 1]
```diff
@@ -61,21 +61,50 @@ apt install -y libmariadb3 libmariadb-dev
...
-cp -R ./include/mariadb/conncpp/compat/* /usr/include/mariadb/conncpp/compat
-
-install -d /usr/lib/mariadb
-install -d /usr/lib/mariadb/plugin
-
-#Shared libraries copied into usr/lib
-cp lib/mariadb/libmariadbcpp.so /usr/lib
-cp -R lib/mariadb/plugin/* /usr/lib/mariadb/plugin
+
+if [ "$(dpkg --print-architecture)" = "arm64" ] ; then
```
Comment (juzuluag) [inline containers/cpp-mariadb/.devcontainer/install-mariadb.sh:88]: @dnastrain , It seems like there is room from improvement here, given the fact that both scenarios shared similar calls. (i.e `cd ${TMP_DIR}`, copy headers to `/usr/include/mariadb` ...).

File: containers/cpp-mariadb/.devcontainer/install-mariadb.sh [hunk 2]
```diff
@@ -61,21 +61,50 @@ apt install -y libmariadb3 libmariadb-dev
...
+    # Instructions are copied and modified from: https://github.com/mariadb-corporation/mariadb-connector-cpp/blob/master/BUILD.md
+    # and from: https://mariadb.com/docs/clients/mariadb-connectors/connector-cpp/install/
+    cd ${TMP_DIR}
+    apt-get update
+    apt-get install -y git cmake make gcc libssl-dev
```
Comment (juzuluag) [inline containers/cpp-mariadb/.devcontainer/install-mariadb.sh]: Are these tools only needed for building the connector? Does it makes sense to remove them after?

**Review by juzuluag — COMMENTED** [review]
@dnastrain, overall, looks good. Just made some comments

File: containers/cpp-mariadb/.devcontainer/install-mariadb.sh (hunk 1 above)
Comment (dnastrain) [reply containers/cpp-mariadb/.devcontainer/install-mariadb.sh:88]: hi @juzuluag, that is true that there are similarities -- was intentionally making an 'arm-only' update here, and while only the _destination_ folders match, the _sources_ of all file copies do differ, so didn't take extra steps to 'merge' them. 
I'll take another look though, for low-hanging-fruit, like the `cd ${TMP_DIR}`

File: containers/cpp-mariadb/.devcontainer/install-mariadb.sh (hunk 2 above)
Comment (dnastrain) [reply containers/cpp-mariadb/.devcontainer/install-mariadb.sh]: well... the pattern that we saw in other scripts, include DCF, was to "apt install, and leave in place" ... will take another look though

**Review by juzuluag — APPROVED** [review]
Great changes. Thx.

**Comment by dnastrain** [comment]
Thx for all the input -- moving this PR over to msft-main repo.

**Pull request closed without merging**